C/services/common/notification_manager.cpp

```diff
@@ -1511,6 +1511,32 @@ bool NotificationManager::auditNotification(const string& notificationName)
  */
 bool NotificationManager::APIdeleteInstance(const string& instanceName)
 {
+	NotificationManager* notifications = NotificationManager::getInstance();
+	NotificationInstance* instance = NULL;
+
+	notifications->lockInstances();
+	instance = notifications->getNotificationInstance(instanceName);
+	notifications->unlockInstances();
+
+	if (instance)
+	{
+		NotificationSubscription* subscriptions = NotificationSubscription::getInstance();
+		string ruleName = instance->getRule()->getName();
+		// Get all assets for this rule
+		std::vector<NotificationDetail>& assets = instance->getRule()->getAssets();
```

Comment:
if two notifications are subscribed to the same asset to you have a subscription for each or do you share it?

Reply:
There are assets for rules and then subscriptions to storage layer per asset

The removeSubscription() takes care of that.

The asset "x" is removed only if it's the only one